File: core/src/com/spacefist/state/CreditsState.java
package com.spacefist.state;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Input.Keys;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.utils.TimeUtils;
import com.spacefist.GameData;
import com.spacefist.state.abst.GameState;

/**
 * This state displays information about the developers of
 * the game.
 */
public class CreditsState implements GameState {
    private GameData gameData;
    private long     enteredAt;

    public CreditsState(GameData gameData) {
        this.gameData = gameData;
    }

    @Override
    public void loadContent() { }

    @Override
    public void draw() {
        Rectangle resolution = gameData.getResolution();

        int y       = ((int) ( resolution.getHeight() - (0.63 * resolution.getHeight())));
        int leftX   = (int) (0.1 * resolution.getWidth());
        int rightX  = (int) (0.7325 * resolution.getWidth());
        int centerX = (int) (0.5 * resolution.getWidth()) - 103;

        gameData.getSpriteBatch().draw(
            gameData.getTextures().get("Credits"),
            0,
            0,
            resolution.getWidth(),
            resolution.getHeight()
        );

        BitmapFont font = gameData.getFont();
        SpriteBatch spriteBatch = gameData.getSpriteBatch();

        font.draw(spriteBatch, "Dongcai Huang", leftX, y);
        font.draw(spriteBatch, "Programming", leftX, y - 30);
        font.draw(spriteBatch, "Art Selection", leftX, y - 45);

        font.draw(spriteBatch, "Tatsuya Takahashi", rightX, y);
        font.draw(spriteBatch, "Programming", rightX + 40, y - 30);

        font.draw(spriteBatch, "Andrew Smith", centerX, y);
        font.draw(spriteBatch, "Programming / AI", centerX - 30, y - 30);
        font.draw(spriteBatch, "Art Selection", centerX, y - 45);
        font.draw(spriteBatch, "Sound Selection", centerX - 10, y - 60);
    }

    @Override
    public void update() {

        if (TimeUtils.millis() - enteredAt > 300)
        {
            enteredAt = TimeUtils.millis();

            boolean mousePressed = Gdx.input.isTouched();
            boolean isEnterDown = Gdx.input.isKeyPressed(Keys.ENTER);
            boolean isEscapeDown = Gdx.input.isKeyPressed(Keys.ESCAPE);

            if (mousePressed || isEnterDown || isEscapeDown) {
                gameData.setCurrentState(gameData.getMenuState());
            }
        }

    }

    @Override
    public void enteringState() {
        enteredAt = TimeUtils.millis();
    }

    @Override
    public void exitingState() { }
}

File: core/src/com/spacefist/state/CreditsState.kt
package com.spacefist.state

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Input.Keys
import com.badlogic.gdx.utils.TimeUtils
import com.spacefist.GameData
import com.spacefist.state.abst.GameState

/**
 * This state displays information about the developers of
 * the game.
 */
class CreditsState(private val gameData: GameData) : GameState {
    private var enteredAt: Long = 0

    override fun loadContent() {}

    override fun draw() {
        val resolution = gameData.resolution

        val y = (resolution.getHeight() - 0.63 * resolution.getHeight()).toInt()
        val leftX = (0.1 * resolution.getWidth()).toInt()
        val rightX = (0.7325 * resolution.getWidth()).toInt()
        val centerX = (0.5 * resolution.getWidth()).toInt() - 103

        gameData.spriteBatch.draw(
                gameData.textures["Credits"],
                0f,
                0f,
                resolution.getWidth(),
                resolution.getHeight()
        )

        val font = gameData.font
        val spriteBatch = gameData.spriteBatch

        font.draw(spriteBatch, "Dongcai Huang", leftX.toFloat(), y.toFloat())
        font.draw(spriteBatch, "Programming", leftX.toFloat(), (y - 30).toFloat())
        font.draw(spriteBatch, "Art Selection", leftX.toFloat(), (y - 45).toFloat())

        font.draw(spriteBatch, "Tatsuya Takahashi", rightX.toFloat(), y.toFloat())
        font.draw(spriteBatch, "Programming", (rightX + 40).toFloat(), (y - 30).toFloat())

        font.draw(spriteBatch, "Andrew Smith", centerX.toFloat(), y.toFloat())
        font.draw(spriteBatch, "Programming / AI", (centerX - 30).toFloat(), (y - 30).toFloat())
        font.draw(spriteBatch, "Art Selection", centerX.toFloat(), (y - 45).toFloat())
        font.draw(spriteBatch, "Sound Selection", (centerX - 10).toFloat(), (y - 60).toFloat())
    }

    override fun update() {

        val timeSinceEnter = TimeUtils.millis() - enteredAt
        val safeToProcessInput = timeSinceEnter > 300

        if (safeToProcessInput) {
            enteredAt = TimeUtils.millis()

            val mousePressed = Gdx.input.isTouched
            val isEnterDown = Gdx.input.isKeyPressed(Keys.ENTER)
            val isEscapeDown = Gdx.input.isKeyPressed(Keys.ESCAPE)

            if (mousePressed || isEnterDown || isEscapeDown) {
                gameData.currentState = gameData.menuState
            }
        }

    }

    override fun enteringState() {
        enteredAt = TimeUtils.millis()
    }

    override fun exitingState() {}
}
